test_fms/mpp/test_mpp_init_logfile.F90

```diff
@@ -0,0 +1,44 @@
+!***********************************************************************
+!*                   GNU Lesser General Public License
+!*
+!* This file is part of the GFDL Flexible Modeling System (FMS).
+!*
+!* FMS is free software: you can redistribute it and/or modify it under
+!* the terms of the GNU Lesser General Public License as published by
+!* the Free Software Foundation, either version 3 of the License, or (at
+!* your option) any later version.
+!*
+!* FMS is distributed in the hope that it will be useful, but WITHOUT
+!* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+!* FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+!* for more details.
+!*
+!* You should have received a copy of the GNU Lesser General Public
+!* License along with FMS.  If not, see <http://www.gnu.org/licenses/>.
+!***********************************************************************
+
+!> @file
+!! @brief Tests the mpp_init_logfile interface
+!! @author Miguel Zuniga
+!! @email gfdl.climate.model.info@noaa.gov
+
+program test_mpp_init_logfile
+#include <fms_platform.h>
+
+  use mpp_mod, only : mpp_init, mpp_init_logfile
+  use mpp_mod, only : mpp_init_test_logfile_init
+
+  IMPLICIT NONE
+
+  integer :: my_mpp_init_level
+  integer :: err_no
+
+  !Initialize mpp but have mpp_init() exit before log files are initalized.
+  my_mpp_init_level = mpp_init_test_logfile_init - 1
+  call mpp_init( test_level = my_mpp_init_level )
+  
+  call mpp_init_logfile()
```

Comment:
is anything actually tested, or is this routine just called?

Reply:
Most of the test work is actually done is done by the shell command calling  this the fortran test program. Like many I/o routines, mpp_init_logfile() is merely a side_effect routine. So what the shell comman does is test that it makes it side effects correctly and also that it doesn't make additional side effects that it could potentially make. This means, checking the files that it should replace, and checking for the lack of existence of files that it should not make but that whose names are iterated over in the do loop of the routine. If you look at the shell command that calls it, you should be able to see that it return a '1' when those conditions are not satisfied. (PS I checked that the CI summary shows "FAILED" is the shell command returns 1, and success if it returns 0.)

Reply:
I updated the comments in both the Fortran test program and the corresponding  shell script. It better explains what they are doing together.